centreon/www/front_src/src/ResourceAccessManagement/AddEditResourceAccessRule/FormInputs/hooks/useDatasetFilter.ts

```diff
@@ -214,14 +298,72 @@ const useDatasetFilter = (
         resources
       );
       setFieldTouched(`datasetFilters.${datasetFilterIndex}`, true, false);
+      setSelectedDatasetFiltes(
+        selectedDatasetFilters.map((datasetF, indexF) => {
+          if (equals(indexF, datasetFilterIndex)) {
+            return selectedDatasetFilters[indexF].map((dataset, i) => {
+              if (equals(i, index)) {
+                return {
+                  ids: pluck('id', resources) as Array<number>,
+                  type: dataset.type
+                };
+              }
+
+              return dataset;
+            });
+          }
+
+          return datasetF;
+        })
+      );
```

Comment:
Same comment here. btw, the logic looks to be the same whenever you set the selectedDatasetFilters, can you write a function in order to avoid code duplication?